admin-dev/themes/new-theme/js/pages/order/view/order-view-page.js

```diff
@@ -0,0 +1,216 @@
+        $(OrderViewPageMap.productsTableRow(event.orderDetailId)),
+        event.newRow
+      );
+      this.orderProductRenderer.resetEditRow(event.orderDetailId);
+      this.orderPricesRefresher.refresh(event.orderId);
+      this.listenForProductDelete();
+      this.listenForProductEdit();
+
+      const editRowsLeft = $(OrderViewPageMap.productEditRow).not(OrderViewPageMap.productEditRowTemplate).length;
+      if (editRowsLeft > 0) {
+        return;
+      }
+      this.orderProductRenderer.moveProductPanelToOriginalPosition();
+    });
+
+    EventEmitter.on(OrderViewEventMap.productAddedToOrder, (event) => {
+      const $tablePagination = $(OrderViewPageMap.productsTablePagination);
+      const numRowsPerPage = $tablePagination.data('numPerPage');
+      const numProducts = parseInt($(OrderViewPageMap.productsCount).html(), 10);
+
+      this.orderProductRenderer.addOrUpdateProductToList(
+        $(`#${$(event.newRow).find('tr').attr('id')}`),
+        event.newRow
+      );
+      this.listenForProductDelete();
+      this.listenForProductEdit();
+
+      if ($(OrderViewPageMap.productsTable).find('tr[id^="orderProduct_"]:not(.d-none)').length >= numRowsPerPage) {
+        // Update pagination
+        let numPages = $tablePagination.data('numPages');
+        if (numProducts % numRowsPerPage === 0) {
+          numPages += 1;
+          this.orderProductRenderer.paginationAddPage(numPages);
+        }
+        // Move to last page
+        EventEmitter.emit(OrderViewEventMap.productListPaginated, {
+          orderId: event.orderId,
+          numPage: numPages
+        });
+      }
+      this.orderProductRenderer.updateNumProducts(numProducts + 1);
+      this.orderProductRenderer.resetAddRow();
+      this.orderPricesRefresher.refresh(event.orderId);
+      this.orderInvoicesRefresher.refresh(event.orderId);
+      $(OrderViewPageMap.addProductNewInvoiceElement).addClass('d-none');
+      this.orderProductRenderer.moveProductPanelToOriginalPosition();
+    });
+  }
+
+  listenForProductDelete() {
+    $(OrderViewPageMap.productDeleteBtn)
+      .off('click')
+      .on('click', event => this.orderProductManager.handleDeleteProductEvent(event));
+  }
+
+  listenForProductEdit() {
+    $(OrderViewPageMap.productEditBtn).off('click').on('click', (event) => {
+      const $btn = $(event.currentTarget);
+      this.orderProductRenderer.moveProductsPanelToModificationPosition();
+      this.orderProductRenderer.editProductFromList(
+        $btn.data('orderDetailId'),
+        $btn.data('productQuantity'),
+        $btn.data('productPriceTaxIncl'),
+        $btn.data('productPriceTaxExcl'),
+        $btn.data('taxRate'),
+        $btn.data('location'),
+        $btn.data('availableQuantity'),
+        $btn.data('orderInvoiceId'),
+      );
+    });
+  }
+
+  listenForProductAdd() {
+    $(OrderViewPageMap.productAddBtn).on(
+      'click',
+      event => {
+        if ($(OrderViewPageMap.productAddInvoiceSelect).val() == 0) {
```

Comment:
This block could be extracted into a function `toggleAddProductNewInvoiceElement` (better naming highly possible ^^)
```js
if ($(OrderViewPageMap.productAddInvoiceSelect).val() == 0) {
    $(OrderViewPageMap.addProductNewInvoiceElement).removeClass('d-none');
} else {
    $(OrderViewPageMap.addProductNewInvoiceElement).addClass('d-none');
}
```

The method should be inside the component `this.orderProductRenderer` This way all the display code is in this component